contrib/go/src/python/pants/contrib/go/tasks/go_buildgen.py

```diff
@@ -88,6 +90,10 @@ def _generate_missing(self, gopath, local_address, import_listing, visited):
                             remote_root, import_path
                         )
                         name = remote_pkg_path or os.path.basename(import_path)
+                        if not self._remote_source_root:
+                            raise self.RemoteRequired(
+                                "You must provide the remote source root with --remote-root"
```

Comment:
This won't actually be that helpful. Perhaps pass in self.options_scope from GoBuildgen to improve the message or just have GoBuildgen pass in a factory function for the remote_source_root that can raise when called if unset to keep all the info local there.

Reply:
Done.